src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker.cs

```diff
@@ -0,0 +1,499 @@
+
+                var idsAndFileInfos = new List<(ProjectId id, ProjectFileInfo fileInfo)>();
+
+                foreach (var projectFileInfo in projectFileInfos)
+                {
+                    var projectId = _projectMap.GetOrCreateProjectId(projectFileInfo);
+
+                    if (_projectIdToFileInfoMap.ContainsKey(projectId))
+                    {
+                        // There are multiple projects with the same project path and output path. This can happen
+                        // if a multi-TFM project does not have unique output file paths for each TFM. In that case,
+                        // we'll create a new ProjectId to ensure that the project is added to the workspace.
+
+                        _diagnosticReporter.Report(
+                            DiagnosticReportingMode.Log,
+                            string.Format(WorkspaceMSBuildResources.Found_project_with_the_same_file_path_and_output_path_as_another_project_0, projectFileInfo.FilePath));
+
+                        projectId = ProjectId.CreateNewId(debugName: projectFileInfo.FilePath);
+                    }
+
+                    idsAndFileInfos.Add((projectId, projectFileInfo));
+                    _projectIdToFileInfoMap.Add(projectId, projectFileInfo);
+                }
+
+                // If this project resulted in more than a single project, a discrimator (e.g. TFM) should be
+                // added to the project name.
+                var addDiscriminator = idsAndFileInfos.Count > 1;
+
+                foreach (var (id, fileInfo) in idsAndFileInfos)
+                {
+                    var projectInfo = await CreateProjectInfoAsync(fileInfo, id, addDiscriminator, cancellationToken).ConfigureAwait(false);
+
+                    builder.Add(projectInfo);
+                    _projectMap.AddProjectInfo(projectInfo);
+                }
+
+                results = builder.ToImmutable();
+
+                _pathToDiscoveredProjectInfosMap.Add(projectPath, results);
+
+                return results;
+            }
+
+            private Task<ProjectInfo> CreateProjectInfoAsync(ProjectFileInfo projectFileInfo, ProjectId projectId, bool addDiscriminator, CancellationToken cancellationToken)
+            {
+                var language = projectFileInfo.Language;
+                Debug.Assert(IsSupportedLanguage(language));
+
+                var projectPath = projectFileInfo.FilePath;
+
+                var projectName = Path.GetFileNameWithoutExtension(projectPath);
+                if (addDiscriminator && !string.IsNullOrWhiteSpace(projectFileInfo.TargetFramework))
+                {
+                    projectName += "(" + projectFileInfo.TargetFramework + ")";
+                }
+
+                var version = GetProjectVersion(projectPath);
+
+                if (projectFileInfo.IsEmpty)
+                {
+                    var assemblyName = GetAssemblyNameFromProjectPath(projectPath);
```

Comment:
is expected thing? or error case? any reason it must be handled explicitly rather than regular one below can handle?

Reply:
This is derived from the current MSBuildWorkspace behavior. This happens if the project fails to build for some reason. In that case, it still gets added to the workspace, but it's contents are mostly empty.